docs/book/src/developer/testing.md

```diff
@@ -70,6 +70,17 @@ Additionally, `test-e2e` target supports the following env variables:
 - `USE_EXISTING_CLUSTER` to use an existing management cluster instead of creating a new one for each test run (default to false)
 - `GINKGO_NOCOLOR` to turn off the ginko colored output (default to false)
 
+### Running specific tests
+
+To run a subset of tests, the `GINKGO_FOCUS` env variable can be set to one of the following values:
+- `[PR-Blocking]` => Sanity tests run before each PR merge
+- `[Feature]` => Tests which verify specific CAPI behaviors
```

Comment:
I think I got what you are aiming here, but [Feature] isn't really intuitive

I'm wondering if we should drop this and instead support GINKGO_SKIP so

- capi-pr-e2e-main has FOCUS=[PR-Blocking] (only quick start)
- capi-pr-e2e-main-full has SKIP=[PR-Blocking] and [K8s-Upgrade] (everything except quick start and K8S upgrades, given that it already tests KCP upgrades, MD upgrades, MP upgrades and conformance separately)
- capi-e2e-main has SKIP=[K8s-Upgrade] (everything except K8S upgrades)
- capi-e2e-main-upgrade has FOCUS=[K8s-Upgrade] (only KCP upgrades)

What do you think?

Reply:
This is similar to what Warren has mentioned below, around some tests not being labeled which makes this idea a bit unclear. I like this idea better around using a combination of SKIP and FOCUS to do the needful.